src/main/scala/edu/arizona/sista/learning/Dataset.scala

```diff
@@ -338,6 +344,32 @@ class RVFDataset[L, F] (
   }
 }
 
+class InformationGain( var datumCount:Int = 0,
+                       val datumsByClass:Counter[Int] = new Counter[Int]) {
+  def ig(total:InformationGain):Double = {
+    var pos = 0.0
+    var neg = 0.0
+    if(pWith(total) != 0) {
+      for (c <- datumsByClass.keySet) {
+        val p = datumsByClass.getCount(c) / datumCount.toDouble
+        pos += p * math.log(p)
+      }
+      pos *= pWith(total)
+    }
+    if(pWithout(total) != 0) {
+      for(c <- total.datumsByClass.keySet) {
+        val p = (total.datumsByClass.getCount(c) - datumsByClass.getCount(c)) / (total.datumCount - datumCount).toDouble
+        neg += p * math.log(p)
+      }
+      neg *= pWithout(total)
+    }
+    pos + neg
+  }
+
+  def pWith(total:InformationGain) = datumCount.toDouble / total.datumCount.toDouble
+  def pWithout(total:InformationGain) = (total.datumCount - datumCount).toDouble / total.datumCount.toDouble
+}
+
```

Comment:
It is incremented somewhere else. This is one of the few mutable objects that should be mutable...